### PR TITLE
validate end_learning_rate and warmup targets in lr_decay_schedulers

### LINKED ISSUE #18747: Question on end_learning_rate and warmup_target of PolynomialDecay and CosineDecay respectively
As per documentation of `PolynomialDecay`, the purpose of this scheduler is to monotonically decrease the learning rate, whose degree of change is carefully chosen, can result in a better performing model. As per my understanding this scheduler is meant to given a provided `initial_learning_rate`, to reach an `end_learning_rate` in the given `decay_steps`. But in current implementation if we provide `end_learning_rate` greater than `initial_learning_rate` the scheduler will increase the learning_rate to `end_learning_rate` which is not the intended use of this decay scheduler and it also affects model performance. Hence it is appropriate to throw an exception for this case.

Similarly for `Cosinedecay` scheduler, we have an option of warmup, purpose of which is increasing from `initial_learning_rate` to `warmup_target` for a duration of `warmup_steps`. But here also if we provide `warmup_target`  which is lower than `initial_learning_rate` instead of increasing learning rate, which is the purpose of warmup, it actually decreasing the learning rate as there is no validation for this. 

Attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/71a8d1a55e1e9a8053f3aeadfc03e3a5/keras3_polynomialdecay-cosinedecay.ipynb) for the demo described above.

Hence proposing validation for both of the above schedulers to make sure they are used for intended purpose only.

### PR DESCRIPTION
**Pull request by SuryanarayanaY**
As per documentation of `PolynomialDecay`, the purpose of this scheduler is to monotonically decrease the learning rate, whose degree of change is carefully chosen, can result in a better performing model. As per my understanding this scheduler is meant to given a provided `initial_learning_rate`, to reach an `end_learning_rate` in the given `decay_steps`. But in current implementation if we provide `end_learning_rate` greater than `initial_learning_rate` the scheduler will increase the `learning_rate` to `end_learning_rate` which is not the intended use of this decay scheduler and it also affects model performance. Hence it is appropriate to throw an exception for this case.

Similarly for `Cosinedecay` scheduler, we have an option of `warmup`, purpose of which is increasing from `initial_learning_rate` to `warmup_target` for a duration of `warmup_steps`. But here also if we provide `warmup_target` which is lower than `initial_learning_rate` instead of increasing learning rate, which is the purpose of warmup, it actually decreasing the learning rate as there is no validation for this.

Attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/71a8d1a55e1e9a8053f3aeadfc03e3a5/keras3_polynomialdecay-cosinedecay.ipynb) for the demo described above.

Hence proposing validation for both of the above schedulers to make sure they are used for intended purpose only.

Fixes #18747

### CODE REVIEW

**Comment by fchollet** [comment]
Mathematically, the schedulers works in any case, right? I don't think it's a good idea to introduce artificial constraints like this.

**Comment by SuryanarayanaY** [comment]
@fchollet , Yeah, Mathematically its working very much fine. My intention is as the name itself suggests its a decay scheduler but it can increase the learning rate instead of decaying.Our documentation of `PolynomialDecay` states it like below which may conflict current implementation as we are particularly mentioning `decay` of learning rate.

```
    It is commonly observed that a monotonically decreasing learning rate, whose
    degree of change is carefully chosen, results in a better performing model.
    This schedule applies a polynomial decay function to an optimizer step,
    given a provided `initial_learning_rate`, to reach an `end_learning_rate`
    in the given `decay_steps`.
```

If a user mishandled this, the exception can save his time and resources.

**Comment by fchollet** [comment]
Using the API to increase rather than decrease the LR can be a valid use case -- let's leave it as is.
